components/admin/metadata/form/SourcesContentModal.js

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { toastr } from 'react-redux-toastr';
+
+// redux
+import { connect } from 'react-redux';
+
+// redactions
+import { setTmpSources, setSources } from 'redactions/admin/sources';
+
+// components
+import ContentGroup from 'components/ui/ContentGroup';
+import Source from 'components/admin/metadata/form/Source';
+import { SOURCE_ELEMENTS } from 'components/admin/metadata/form/constants';
+
+class SourcesContentModal extends React.Component {
+
+  componentWillMount() {
+    const { sources, setTmpSources } = this.props;
```

Comment:
Try a dataset that doesn't have any source added. Could you please set the initial state as an array with one empty object? Just the same you are doing when you try to remove all sources...

![image](https://user-images.githubusercontent.com/8928965/30959166-a30009ae-a43f-11e7-9ee7-96b065fc5879.png)

Reply:
Also, could you list the selected sources in the metadata form? At least string separated by commas

Reply:
![image](https://user-images.githubusercontent.com/999124/31067319-8a1a3eee-a752-11e7-860c-3daf60254e2e.png)

Added source list.